x-pack/packages/ml/url_state/src/url_state.tsx

```diff
@@ -239,7 +232,7 @@ export class PageUrlStateService<T> {
  * Hook for managing the URL state of the page.
  */
 export const usePageUrlState = <PageUrlState extends object>(
-  pageKey: AppStateKey,
+  pageKey: string,
```

Comment:
I'd like to avoid having too plugin specific code in the packages which are used by multiple plugins. Instead I now did it the other way around and changed the required generic for `usePageUrlState` to be a combo of the `pageKey` definition and the original url state interface in 2f84808. Hope that's good enough for this iteration.